src/app/src/main/kotlin/org/vaccineimpact/orderlyweb/OrderlyServerClient.kt

```diff
@@ -0,0 +1,40 @@
+package org.vaccineimpact.orderlyweb
+
+import okhttp3.OkHttpClient
+import org.vaccineimpact.orderlyweb.db.Config
+import org.vaccineimpact.orderlyweb.errors.PorcelainError
+import org.vaccineimpact.orderlyweb.models.Parameter
+
+interface OrderlyServerAPI : PorcelainAPI
+{
+    @Throws(PorcelainError::class)
+    fun getRunnableReportNames(queryParams: Map<String, String>): List<String>
+
+    @Throws(PorcelainError::class)
+    fun getReportParameters(reportName: String, queryParams: Map<String, String>): List<Parameter>
+
+    override fun throwOnError(): OrderlyServerAPI
+}
+
+class OrderlyServerClient(
+        config: Config,
+        client: OkHttpClient = OkHttpClient()
+) : OrderlyServerAPI, PorcelainAPIClient("Orderly server", config["orderly.server"], client)
```

Comment:
The change here is that all the basic verb functionality (get, post, etc) comes from `PorcelainAPIClient`, and only the 2 custom endpoint methods remain in this class. No logic has actually changed.

Reply:
IIRC these were introduced because these 2 methods are called multiple times, so were given special methods to avoid duplication. I have no strong feelings about this personally.